sfs/mono/wfs.py

```diff
@@ -185,6 +191,16 @@ def point_25d(omega, x0, n0, xs, xref=[0, 0, 0], c=None, omalias=None):
 
     Notes
     -----
+    `point_25d_legacy` originates from deriving 2.5D WFS from the 2D
+    Neumann-Rayleigh integral (i.e. the approach by Rabenstein & Spors), cf.
+    Spors, S.; Rabenstein, R.; Ahrens, J. (2008): The theory of Wave Field
+    Synthesis revisited. In: Proc. of 124th Audio Eng. Soc. Convention,
+    Amsterdam, #7358.
+    Also cf. Eq. (2.145)-(2.147) in https://doi.org/10.18453/rosdok_id00001765
```

Comment:
Please use our BibTeX-based system for references.

Reply:
Ok, I will do it. However, none of the existing driving filters/signals use this handling at the moment. Can you provide an example?
Besides that, I'd strongly suggest always adding a DOI or other permalink of the reference directly in the docstring help for users that only use the source code. This helps a lot, since we can not always guarantee that our referencing docu is correct, or can we indeed?

Reply:
> Can you provide an example?

Sure: 

https://github.com/sfstoolbox/sfs-python/blob/a15b4f362989493597e7f381ed2d9ec23b1d08e3/sfs/mono/wfs.py#L250-L252

and

https://github.com/sfstoolbox/sfs-python/blob/a15b4f362989493597e7f381ed2d9ec23b1d08e3/doc/references.bib#L41-L46

> I'd strongly suggest always adding a DOI or other permalink of the reference directly in the docstring help for users that only use the source code.

We should either do everything manually or use a centralized approach. We should definitely not do both at the same time.
I think the centralized approach makes much more sense.

> we can not always guarantee that our referencing docu is correct, or can we indeed?

I don't know what you mean.

But if you read ``:cite:`Spors2008` `` in the source code, you can go to `doc/references.bib`, search for "Spors2008" and get all the information you need.

Reply:
At the beginning I was also sceptical regarding the central approach and prefered ASCII math over proper LaTeX formula as ASCII was easier to read directly in the code. But in the end I realized that I was always using the online documentation to read the docstrings anyway.

I would also prefer to use something like `` :cite:`Spors2008` `` in the docstrings.

Reply:
> But if you read `` :cite:`Spors2008` `` in the source code, you can go to `doc/references.bib`, search for "Spors2008" and get all the information you need.

I'm not convinced. The code itself should contain a reference (DOI is enough, no fancy ASCII equation art) what actually is implemented with no need for a further file. This might change sometimes and nobody cares for the back link. Or the file might be not at hand, or or or...
I don't see a flaw to give just one more URL in the docstring additionally to the bibtex entry, which by its nature is fail safe, the other approach not!

Reply:
Thanks for the bib example, works charming! I've added the references.

Reply:
It's true that in the case where you have previously installed the SFS Toolbox with `pip` and you don't have an internet connection, you will not be able to access the `references.bib` file (because it's not part of the wheel package).

But the DOI doesn't help either, because you don't have an internet connection, remember?

If you *do* have an internet connection, you should really just go to https://sfs-python.readthedocs.io/ and look at the documentation.

And if you know that you'll be offline at some point, you can get the PDF docs, which also include the references.
Or you can build the HTML docs locally.